spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -113,17 +112,67 @@ public ParameterDescription describe(MethodParameter parameter) {
 
 	@Override
 	public List<CompletionProposal> complete(MethodParameter parameter, CompletionContext context) {
-		return null;
+		// DevNote: in Legacy the user is either completing an argument (-- prefix) or the value of the argument (e.g. no arity to consider).
+		// Other cases to be considered include default keys and default values
+		
+		CliOption option = parameter.getParameterAnnotation(CliOption.class);
+		List<String> keys = Arrays.asList(option.key());
+		
+		int currentWordIndex = context.getWordIndex();		
+		
+		// 1) has a key for this parameter already been defined?
+		boolean keyAlreadyDefined = context.getWords().stream()
+				.filter(word -> word.startsWith(CLI_PREFIX) && keys.contains(word.substring(CLI_PREFIX.length())))
+				.map(word -> true)
+				.findFirst().orElse(false);
+		
+		if (keyAlreadyDefined) {
+			// if a key has already been provided for this parameter, then there is no need to provide completion
+			// proposals
+			return Collections.emptyList();
+		}
+		
+		// 2) is the user currently completing an argument key?
+		String currentWordUpToCursor = context.currentWordUpToCursor();
+		if (currentWordUpToCursor != null && currentWordUpToCursor.startsWith(CLI_PREFIX)) {
+			// return the keys of this parameter that aren't empty and match the current key being completed
+			return keys.stream()
+					.filter(key -> !key.isEmpty() && (CLI_PREFIX + key).startsWith(currentWordUpToCursor))
+					.map(key -> completionProposalFor(parameter, option, key))
+					.collect(Collectors.toList());
+		}
+		
+		// 3) is the previous word an argument key?
+		boolean isPreviousWordArgumentKey = currentWordIndex > 0 && context.getWords().get(currentWordIndex-1).startsWith(CLI_PREFIX);
+		
+		if(isPreviousWordArgumentKey) {
+			// in Shell 1 we don't provide value completions, and if the previous word is a key then
```

Comment:
What do you mean we don't provide value completions? We definitely do! This is why I was amazed not to see `Converter` in you list of imports.

See `Converter::getAllPossibleValues()` from Shell1